### PR TITLE
Use start+length as end for insertions when querying loqusdb. Fix #1743

### LINKED ISSUE #1743: Missed freq for insertions
Daniel,
https://scout.scilifelab.se/cust002/F0032963/sv/variants

https://scout.scilifelab.se/cust002/F0026375/sv/variants

PYGM 75 nt

### PR DESCRIPTION
**Pull request by moonso**
This PR fixes a problem with local frequencies for INSertions. See #1743 

**Prepare for tests**:
1. - [x] Go to a case with a known insertion. Check that there are no local observations:
Skärmavbild 2020-02-21 kl  15 16 52

**How to test**:
1. - [x] Install on clinical-db
1. - [x] Refresh and check that there are observations
1. - [x] Upload a screendump

**Expected outcome**:
The observations should be shown

**Review:**
- [x] code approved by
- [x] tests executed by @moonso

### CODE REVIEW

**Comment by moonso** [comment]
Skärmavbild 2020-02-21 kl  15 21 37

**Comment by moonso** [comment]
Done and done @dnil @northwestwitch

**Comment by dnil** [comment]
Ok! Nice! How about the ones that does not have a "length" (n.d.)?  Should be fine I guess, but testing a fair number from stage finds none with frequency,
https://scout-stage.scilifelab.se/cust000/F0005756-mip7/sv/variants/340045bb440a8cdbe9bb907810538d69
whereas I quickly come across ones in prod scout - eg 
https://scout.scilifelab.se/cust002/F0032970/sv/variants/5ce064281ebe6dc65167f6af6adccb3d
I haven't checked the db, but perhaps we need a check against length=-1?

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,19 +8,15 @@ About changelog [here](https://keepachangelog.com/en/1.0.0/)
...
-### Added
-### Fixed
-- Layout on SV page when local observations without cases are present
 ### Changed
 
 ## [4.12.3]
 
 ### Fixed
 - Case report when causative or pinned SVs have non null allele frequencies
+- Layout on SV page when local observations without cases are present
```
[the remark this answers is not in the document]
Comment (moonso) [reply CHANGELOG.md]: Hehe oops, to late

File: scout/server/extensions.py [hunk 2]
```diff
@@ -115,20 +115,26 @@ def _fetch_variant(self, variant_info):
         variant_call.extend(["variants", "--to-json", "--variant-id", loqus_id])
         # If sv we need some more info
         if variant_info.get("category", "snv") in ["sv"]:
+            sv_type = variant_info["variant_type"]
+            start = variant_info["pos"]
+            end = variant_info["end"]
+            # Insertions have same start and end positions
+            if sv_type == "INS":
+                end = start + variant_info["length"]
```
[the remark this answers is not in the document]
Comment (moonso) [reply scout/server/extensions.py]: I'm not sure, what does that even mean?

File: CHANGELOG.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (dnil) [reply CHANGELOG.md]: You will have to rebase the changelog! 😜

File: scout/server/extensions.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (dnil) [reply scout/server/extensions.py]: I suppose insertion variant with undetermined size - typically larger than library insert/fragment size?

[the remark this answers is not in the document]
Comment (moonso) [reply scout/server/extensions.py]: Ok good point. I think we do what we can in that situation and only use pos and end which will probably be the same. But I should add a check for that in the code before merging. It also starts to look like we need to add some testing.

[the remark this answers is not in the document]
Comment (moonso) [reply scout/server/extensions.py]: Ok I added a clause for this situation

**Comment by moonso** [comment]
Hi @dnil , I fixed the things we discussed and added tests. Could you finish the review?

File: scout/server/app.py [hunk 3]
```diff
@@ -1,21 +1,25 @@
...
-try:
-    from urllib.parse import unquote
-except ImportError:
-    from urllib2 import unquote
-
-from pprint import pprint as pp
-
 import coloredlogs
-from flask import current_app, Flask, redirect, request, url_for
+from flask import Flask, current_app, redirect, request, url_for
```
[the remark this answers is not in the document]
Comment (moonso) [reply scout/server/app.py]: You know black...

File: scout/server/blueprints/variant/controllers.py [hunk 4]
```diff
@@ -271,7 +256,9 @@ def observations(store, loqusdb, case_obj, variant_obj):
 
     obs_data["cases"] = []
     institute_id = variant_obj["institute"]
-    for case_id in obs_data.get("families", []):
+    for i, case_id in enumerate(obs_data.get("families", [])):
+        if i > 10:
```
[the remark this answers is not in the document]
Comment (moonso) [reply scout/server/blueprints/variant/controllers.py]: You will still see how many cases it was found in. This is just for the situation where you want to directly access them

[the remark this answers is not in the document]
Comment (dnil) [reply scout/server/blueprints/variant/controllers.py]: Right; but the user won't know that. Perhaps a popover or at least a documentation entry. Also constants like that in the middle of a function? I guess we can find it, but.. 😅

[the remark this answers is not in the document]
Comment (moonso) [reply scout/server/blueprints/variant/controllers.py]: Ok I added in documentation and a tooltip. If there is a need to have a variable constant here we can add that later.

File: scout/server/blueprints/variant/templates/variant/variant_details.html [hunk 5]
```diff
@@ -118,7 +118,7 @@
       </table>
     </div>
     <div class="card-footer">
-      <strong>Cases</strong>
+      <a href="#" data-toggle="tooltip" title="Show maximum 10 cases"><strong>Cases</strong></a>
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/variant_details.html]: 👍

Comment (moonso) [reply scout/server/blueprints/variant/templates/variant/variant_details.html]: I'll merge this soon

Comment (dnil) [reply scout/server/blueprints/variant/templates/variant/variant_details.html]: No, not with that forcera-pushen Berge.

Comment (dnil) [reply scout/server/blueprints/variant/templates/variant/variant_details.html]: 😂

Comment (dnil) [reply scout/server/blueprints/variant/templates/variant/variant_details.html]: Sorry, check my merge on the changelog. I fixed it for you once, but you forced over it. You are rewriting history... 😉

Comment (moonso) [reply scout/server/blueprints/variant/templates/variant/variant_details.html]: 🤣 it never ends

**Pull request merged**